core/src/cluster_info_vote_listener.rs

```diff
@@ -733,6 +752,8 @@ impl ClusterInfoVoteListener {
                 .or_insert(is_gossip_vote);
         }
 
+        *latest_vote_slot = max(*latest_vote_slot, last_vote_slot);
+
         if is_new_vote {
             subscriptions.notify_vote(*vote_pubkey, vote, vote_transaction_signature);
```

Comment:
Do we need to move this RPC notification before the new check as well so that we still get notified of last voted slots `X` in votes where `X  < latest_vote_slot`

Reply:
This RPC notification is outside the loop, the behavior should remain unchanged. If the last vote was for a new slot as decided by the vote tracker, it will still notify.

Reply:
ah dam the github formatting messed me up, thanks